web-components/src/components/form/BatchBasicsForm.tsx

```diff
@@ -0,0 +1,243 @@
+  start_date: Date | null;
+  end_date: Date | null;
+  metadata?: Partial<BatchMetadataLD>;
+}
+
+const vcsMetadataSchema: AnyObjectSchema = object({
+  'regen:vcsProjectId': string()
+    .matches(numericOnlyRE, { message: invalidVCSID })
+    .required(requiredMessage),
+  'regen:vcsRetirementSerialNumber': string()
+    .matches(vcsRetirementSerialRE, { message: invalidVCSRetirement })
+    .required(requiredMessage),
+});
+
+const validationSchema = object({
+  class_id: string().required(requiredMessage),
+  start_date: date().required(requiredMessage).typeError(invalidDate),
+  end_date: date().required(requiredMessage).typeError(invalidDate),
+  metadata: vcsMetadataSchema,
+});
+
+const BatchBasicsForm: React.FC<{
+  submit: (values: BatchBasicsFormValues) => Promise<void>;
+}> = ({ submit }) => {
+  return (
+    <Formik
+      enableReinitialize
+      validateOnMount
+      initialValues={{
+        class_id: '',
+        start_date: null,
+        end_date: null,
+        metadata: undefined,
+      }}
+      validationSchema={validationSchema}
+      onSubmit={async (values, { setSubmitting }) => {
+        setSubmitting(true);
+        try {
+          await submit(values);
+          setSubmitting(false);
+        } catch (e) {
+          setSubmitting(false);
+        }
+      }}
+    >
+      {({ submitForm, isValid, isSubmitting, touched, values }) => {
+        const isVCS = values?.class_id === 'C01';
+
+        return (
+          <Form>
+            <OnBoardingCard>
+              <Field
+                name="class_id"
+                label="Credit Class"
+                required
+                component={SelectTextField}
+                options={[
+                  { value: '', label: 'Choose Credit Class' },
+                  { value: 'C01', label: 'Verified Carbon Standard (C01)' },
+                ]}
+              />
+              <Box
+                sx={{
+                  display: 'flex',
+                  flexDirection: ['column', 'row'],
+                  justifyContent: 'space-between',
+                  mt: 10,
+                }}
+              >
+                <Box sx={{ mr: [0, 1] }}>
+                  <Field
+                    label="Start Date"
+                    placeholder="Start Date"
+                    name="start_date"
+                    required
+                    component={DatePickField}
+                  />
+                </Box>
+                <Box sx={{ mt: [10, 0], ml: [0, 1] }}>
+                  <Field
+                    label="End Date"
+                    placeholder="End Date"
+                    name="end_date"
+                    required
+                    component={DatePickField}
+                  />
+                </Box>
+              </Box>
+              <Body size="sm" mt={2}>
+                The dates should correspond to the batch monitoring period or
+                vintage.
+              </Body>
+              {isVCS ? (
```

Comment:
for each credit batch metadata (dependent on the credit class), with this pattern, we'll need to hardcode the values we're expecting in the front-end
we only have one credit class for now so that's fine I guess but as soon as we want to have something more maintainable, I believe the way to go would rather be to fetch the SHACL schema from the db based on the credit class (also see how we plan to do that for projects: https://github.com/regen-network/regen-registry-standards/pull/16#discussion_r867876007) and get the fields from it (and also run the metadata validation based on that), which is a bit out of scope of this issue...

Reply:
cc/ @mhagel @wgwz

Reply:
sounds good

Reply:
glad to help further with this, apologies for the delayed response. (tangentially i think i've got a better system for catching up on GH notifs now) thanks for the ping @blushi